x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/population_view/metric_selection.tsx

```diff
@@ -0,0 +1,268 @@
+
+    const cs = {
+      ...defaultChartSettings,
+      intervalMs: interval.getInterval().asMilliseconds(),
+    };
+    if (aggFieldPairList.length > 2) {
+      cs.cols = 3;
+      cs.height = '150px';
+      cs.intervalMs = cs.intervalMs * 3;
+    } else if (aggFieldPairList.length > 1) {
+      cs.cols = 2;
+      cs.height = '200px';
+      cs.intervalMs = cs.intervalMs * 2;
+    }
+    return cs;
+  }
+
+  async function loadCharts() {
+    const cs = getChartSettings();
+    setChartSettings(cs);
+
+    if (aggFieldPairList.length > 0) {
+      const resp: LineChartData = await chartLoader.loadPopulationCharts(
+        jobCreator.start,
+        jobCreator.end,
+        aggFieldPairList,
+        jobCreator.splitField,
+        cs.intervalMs
+      );
+
+      setLineChartsData(resp);
+    }
+  }
+
+  async function loadFieldExamples() {
+    const promises: any[] = [];
+    aggFieldPairList.forEach((af, i) => {
+      if (af.by !== undefined && af.by.field !== null) {
+        promises.push(
+          (async (index: number, field: Field) => {
+            return {
+              index,
+              fields: await chartLoader.loadFieldExampleValues(field),
+            };
+          })(i, af.by.field)
+        );
+      }
+    });
+    const results = await Promise.all(promises);
+    const fieldValues = results.reduce((p, c) => {
+      p[c.index] = c.fields;
+      return p;
+    }, {}) as DetectorFieldValues;
+
+    const newPairs = aggFieldPairList.map((pair, i) => ({
+      ...pair,
+      ...(pair.by === undefined || pair.by.field === null
+        ? {}
+        : {
+            by: {
+              ...pair.by,
+              value: fieldValues[i][0],
+            },
+          }),
+    }));
+    setAggFieldPairList([...newPairs]);
+    setFieldValuesPerDetector(fieldValues);
+  }
+
+  return (
+    <Fragment>
+      {isActive === true && (
+        <Fragment>
+          <SplitFieldSelector />
+          {splitField !== null && <EuiHorizontalRule margin="l" />}
+        </Fragment>
+      )}
+
+      {isActive === false && splitField === null && (
+        <Fragment>
+          Population label TODO
+          {splitField !== null && <EuiHorizontalRule margin="l" />}
+        </Fragment>
+      )}
+
+      {lineChartsData && splitField !== null && (
+        <ChartGrid
+          aggFieldPairList={aggFieldPairList}
+          chartSettings={chartSettings}
+          splitField={splitField}
+          lineChartsData={lineChartsData}
+          modelData={modelData}
+          anomalyData={anomalyData}
+          deleteDetector={isActive ? deleteDetector : undefined}
+          jobType={jobCreator.type}
+          fieldValuesPerDetector={fieldValuesPerDetector}
+        />
+      )}
+      {isActive === true && splitField !== null && (
+        <MetricSelector
+          fields={fields}
+          detectorChangeHandler={detectorChangeHandler}
+          selectedOptions={selectedOptions}
+          maxWidth={560}
```

Comment:
nit: could this become a common constant? it's hard coded twice in two different files.